test/ConfigCatClientTests.ts

```diff
@@ -350,128 +351,222 @@ describe("ConfigCatClient", () => {
+
+    await Promise.all([client.getValueAsync("debug", false), client.getValueAsync("debug", false)]);
+    assert.equal(1, configFetcher.calledTimes);
+  });
+
+  it("Initialization With LazyLoadOptions - multiple getValue calls should not cause multiple config fetches", done => {
+
+    let configFetcher = new FakeConfigFetcher(500);
+    let configCatKernel: FakeConfigCatKernel = { configFetcher };
+    let options: LazyLoadOptions = new LazyLoadOptions("APIKEY");
+    let client: IConfigCatClient = new ConfigCatClient(options, configCatKernel);
+
+    let callbackCount = 0;
+
+    const callback = (value: any) => {
+      {
+        callbackCount++;
+        if (callbackCount > 1) {
+          assert.equal(1, configFetcher.calledTimes);
+          done();
+        }
+      }
+    };
+
+    client.getValue("debug", false, callback);
+    client.getValue("debug", false, callback);
+  });
+
+  it("Initialization With AutoPollOptions with expired cache - getValue should take care of maxInitWaitTimeSeconds", done => {
+
+    let configFetcher = new FakeConfigFetcher(500);
+    let configCache = new FakeCache(new ProjectConfig(new Date().getTime() - 10000000, "{\"f\": { \"debug\": { \"v\": false, \"i\": \"abcdefgh\", \"t\": 0, \"p\": [], \"r\": [] } } }", "etag2"))
+    let configCatKernel: FakeConfigCatKernel = { configFetcher, cache: configCache };
+    let options: AutoPollOptions = new AutoPollOptions("APIKEY", { cache: configCache, maxInitWaitTimeSeconds: 10 });
+    let client: IConfigCatClient = new ConfigCatClient(options, configCatKernel);
+
+    client.getValue("debug", false, value => {
+      done(value == true ? null : new Error("Wrong value."));
+    });
+  });
+
+  it("Initialization With AutoPollOptions with expired cache - getValueAsync should take care of maxInitWaitTimeSeconds", async () => {
+
+    let configFetcher = new FakeConfigFetcher(500);
+    let configCache = new FakeCache(new ProjectConfig(new Date().getTime() - 10000000, "{\"f\": { \"debug\": { \"v\": false, \"i\": \"abcdefgh\", \"t\": 0, \"p\": [], \"r\": [] } } }", "etag2"))
+    let configCatKernel: FakeConfigCatKernel = { configFetcher, cache: configCache };
+    let options: AutoPollOptions = new AutoPollOptions("APIKEY", { cache: configCache, maxInitWaitTimeSeconds: 10 });
+    let client: IConfigCatClient = new ConfigCatClient(options, configCatKernel);
+
+    const value = await client.getValueAsync("debug", false);
+    assert.isTrue(value);
+  });
+
+  it("Dispose should stop the client in every scenario", done => {
+
+    let configFetcher = new FakeConfigFetcher(500);
+    let configCatKernel: FakeConfigCatKernel = { configFetcher };
+    let options: AutoPollOptions = new AutoPollOptions("APIKEY", { pollIntervalSeconds: 2 });
+    let client: IConfigCatClient = new ConfigCatClient(options, configCatKernel);
+    client.dispose();
+    assert.equal(configFetcher.calledTimes, 0);
+    setTimeout(() => {
```

Comment:
IMHO using only one setTimeout can be more readable.


```
setTimeout(() => {
        assert.equal(configFetcher.calledTimes, 1);
        done();
}, 4000);
```

Reply:
truethat